BEIMA.Backend/AuthFunctions/Login.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using BEIMA.Backend.Models;
+using BEIMA.Backend.AuthService;
+using BEIMA.Backend.MongoService;
+using MongoDB.Bson.Serialization;
+using BCryptNet = BCrypt.Net.BCrypt;
+using MongoDB.Bson;
+using System.Collections.Generic;
+
+namespace BEIMA.Backend.AuthFunctions
+{
+    /// <summary>
+    /// Handles a request to login
+    /// </summary>
+    public static class Login
+    {
+        /// <summary>
+        /// Handles a login request.
+        /// </summary>
+        /// <param name="req">The http request.</param>
+        /// <param name="log">The logger to log to.</param>
+        /// <returns>An http response containing a jwt token of the user</returns>
+        [FunctionName("Login")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "login")] HttpRequest req,
+            ILogger log)
+        {
+            List<BsonDocument> userDocs;
+            string password;
+            try
+            {
+                // Deserialize request
+                string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+                var data = JsonConvert.DeserializeObject<LoginRequest>(requestBody);
+                
+                if(data == null || data.Username == null || data.Password == null)
+                {
+                    return new UnauthorizedResult();
+                }
+                password = data.Password;
+
+                // Get all users with request's username
+                var dbServce = MongoDefinition.MongoInstance;
+                var usernameFilter = MongoFilterGenerator.GetEqualsFilter("username", data.Username);
+                userDocs = dbServce.GetFilteredUsers(usernameFilter);
+            } catch (Exception)
+            {
+                return new UnauthorizedResult();
```

Comment:
```suggestion
                return new UnauthorizedResult("Login unauthorized.");
```
We should make sure to return messages with our failures. Also, make sure to add the message to the resx file so that we can keep them all in one place like we are with the other ones.

Reply:
UnauthorizedResults do not have any constructor parameters. I can do this though instead: `return new BadRequestObjectResult(Resources.UnauthorizedMessage) { StatusCode = 401 };`

Reply:
Let's do what Keelan did on his add user endpoint then. This is what he did:
![image](https://user-images.githubusercontent.com/43118853/161401235-157b0b69-3e62-4eb4-9fa9-18c2af5c2948.png)